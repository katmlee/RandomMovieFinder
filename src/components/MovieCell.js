import React from 'react';
import {Text, Image, View} from 'react-native';
// https://image.tmdb.org/t/p/w500/movie/q0R4crx2SehcEEQEkYObktdeFy.jpg

// https://image.tmdb.org/t/p/w500/uX7LXnsC7bZJZjn048UCOwkPXWJ.jpg
const MovieCell = ({movie}) => (
  <View style={styles.cell}>
    <Image style={styles.image} source={{uri: `https://image.tmdb.org/t/p/w500${movie.poster_path}`}} />
    <Text key={movie.id}>{movie.title}</Text>
  </View>
);

const styles = {
  cell: {
    flexDirection: 'row'
  },
  image: {
    width: 50,
    height: 50,
    resizeMode: 'contain'
  }
}

export default MovieCell;
