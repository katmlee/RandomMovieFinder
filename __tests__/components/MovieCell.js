import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MovieCell from '../../src/components/MovieCell';

it('renders correctly', () => {
  const movie = {};
  const tree = renderer.create(
    <MovieCell poster_path="" movie={movie} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
