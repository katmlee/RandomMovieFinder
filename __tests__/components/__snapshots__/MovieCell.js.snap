// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500undefined",
        }
      }
      style={
        Object {
          "height": 50,
          "resizeMode": "contain",
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        /10
      </Text>
    </View>
  </View>
</View>
`;
